### ISSUE TITLE
Installation

### ISSUE THREAD

**Issue by ilpise** [smart-facility/cognicity-server#14]
Hello,
I'm trying to install the cognicity-server on ubuntu 14.04.
I've changed the configuration file in order to connect to localhost database.
When I run `npm test` from inside the cognicity-server directory I've 1 failing with this error.
52 passing (10s)
1 failing

  1) server.js should launch with default configuration and fail with an error:
     Error: timeout of 10000ms exceeded. Ensure the done() callback is being called in this test.
  
npm ERR! cognicity-server@1.0.5 mocha: `mocha test`
npm ERR! Exit status 1
npm ERR! 
npm ERR! Failed at the cognicity-server@1.0.5 mocha script.
npm ERR! This is most likely a problem with the cognicity-server package,
npm ERR! not with npm itself.
npm ERR! Tell the author that this fails on your system:
npm ERR!     mocha test
npm ERR! You can get their info via:
npm ERR!     npm owner ls cognicity-server
npm ERR! There is likely additional logging output above.
npm ERR! System Linux 3.13.0-32-generic
npm ERR! command "/usr/bin/node" "/usr/bin/npm" "run" "mocha"
npm ERR! cwd /home/simone/cognicity-server
npm ERR! node -v v0.10.40
npm ERR! npm -v 1.4.28
npm ERR! code ELIFECYCLE
npm ERR! 
npm ERR! Additional logging details can be found in:
npm ERR!     /home/simone/cognicity-server/npm-debug.log
npm ERR! not ok code 0
npm ERR! Test failed.  See above for more details.
npm ERR! not ok code 0

But I can reach the web page connectin to the 8081 port

If I run the `node server.js config.js` no node process starts at all

**Comment by matthewberryman** [comment]
Regarding the test, can you please attach /home/simone/cognicity-server/test.log so I can check and see more details about why it's failing that test.

I'm not sure why you're able to get to a web site on 8081 given you report no node process is starting. It could be that you already have something else running on port 8081, and then that is causing it to fail. Can you also attach the cognicity-server.log file — the location of where this is put is [defined in the config file](https://github.com/smart-facility/cognicity-server/blob/master/config.js#L104) (note that that directory must be writeable by whatever user you are running node as).

**Comment by ilpise** [comment]
Sorry, my fault, the directory of the log inside the config.js file was not there.
Now the `node server.js config.js` works correctly.
The error about the `npm test` is the same (Using the custom parameters to connect to the database).
The test.log say
`2015-10-13T09:37:05.863Z - info: Application starting, listening on port 8081`

**Comment by matthewberryman** [comment]
There are actually two issues here #15 and #16. I will submit some pull requests for @talltom to review.

**Comment by matthewberryman** [comment]
@ilpise: 
I believe I have fixed the issue you experienced with the log directory (#15), so that the default is to put the log file in the current working directory. See 563abb8374182e6e51e8ebbb62442d14a3860009 for the change of location and 583ed78aaef031efa981dcaab41175d2b7b256a5 for the change in documentation. I would like you to review the documentation before @talltom closes the pull request to merge these changes back in to the master branch. I have separately changed and tested our Amazon Elastic Beanstalk (EB) configuration (see 56fc4e3b4f874b903e9b9cb01691994c91ab6203 if you are interested) so that we can still easily extract log files from our Amazon EC2 instances into Amazon S3. 
@talltom and I had discussed this issue before, but at the time I didn't see a clear path to satisfy both local testing and EB issues with one config.js file, but I do now.

I have fixed the issue you were experiencing with npm test (#16) with #17 and @talltom has merged that back into the master branch. I realised in investigating that issue that there's also a separate issue (also discussed in comments on #16) with running npm test on a system that already has the server running, which I think we should also address. I have addressed that with #19 that @talltom will review and merge.

Thank you so much for raising these issues. Once you've reviewed the documentation and @talltom has had a look as well we'll close off the outstanding pull requests that solve the sub-issues and then I'll close this issue.
